aocdbc/day_4_passport_processing.py

```diff
@@ -0,0 +1,107 @@
+from re import fullmatch
+import sys
+from typing import Dict, List, Optional
+
+from icontract import require, ensure
+
+_REQUIRED_KEYS = set(["byr", "iyr", "eyr", "hgt", "hcl", "ecl", "pid"])
+
```

Comment:
```suggestion
# crosshair: on
```

Then you can delete all `@ensure`?

Reply:
Oh I guess I accidentally misled you here.

Functions without any conditions at all still aren't checked. It used to be the case that you needed a postcondition. Now, a precondition is sufficient (because raised exceptions will be considered errors). But still *some* condition is required. The directives only limit what's eligible for analysis.

Probably we need a higher bandwidth discussion to dig into my reservations about doing some sort of blind analysis, but note this [similar desire](https://github.com/pschanely/CrossHair/issues/22).

Reply:
Let's discuss this off GitHub at the next occasion.